tap_chargebee/client.py

```diff
@@ -12,13 +12,122 @@
 LOGGER = singer.get_logger()
 
 
-class Server4xxError(Exception):
+class ChargebeeError(Exception):
     pass
 
+class Server4xxError(ChargebeeError):
+    pass
+
+class Server429Error(ChargebeeError):
+    pass
+
+class Server5xxError(ChargebeeError):
+    pass
+
+class ChargebeeBadRequestError(Server4xxError):
+    pass
+
+class ChargebeeUnauthorizedError(Server4xxError):
+    pass
 
-class Server429Error(Exception):
+class ChargebeeForbiddenError(Server4xxError):
     pass
 
+class ChargebeeNotFoundError(Server4xxError):
+    pass
+
+class ChargebeeMethodNotAllowedError(Server4xxError):
+    pass
+
+class ChargebeeNotProcessedError(Server4xxError):
+    pass
+
+class ChargebeeRateLimitError(Server429Error):
+    pass
+
+class ChargebeeInternalServiceError(Server5xxError):
+    pass
+
+class ChargebeeServiceUnavailableError(Server5xxError):
+    pass
+
+
+STATUS_CODE_EXCEPTION_MAPPING = {
+    400: {
+        "raise_exception": ChargebeeBadRequestError,
+        "message": "The request URI does not match the APIs in the system.",
+    },
+    401: {
+        "raise_exception": ChargebeeUnauthorizedError,
+        "message": "The user is not authorized to use the API.",
```

Comment:
```suggestion
        "raise_exception": ChargebeeAuthenticationError,
        "message": "The user is not authenticated to use the API.",
```
Ref: https://apidocs.chargebee.com/docs/api?prod_cat_ver=2#error_codes_list